### ISSUE TITLE
Stop rendering waterway=dock as water areas

### ISSUE THREAD

**Issue by imagico** [gravitystorm/openstreetmap-carto#2701]
With respect to #2700 and #2557 i had a look at tagging practice and it seems the vast majority of uses of this tag are for piers and harbor basins.  Actual docks in sense of the purpose of the tag are in the minority by a large margin.

We have the key [dock](https://taginfo.openstreetmap.org/keys/dock#values) with the most common values being `drydock` and `floating` and the latter is of course also widely abused for floating piers.  `drydock` however is used quite consistently.

So i would suggest to drop rendering `waterway=dock`and look into if and how specific values of `dock=*` could be rendered, preferably in a way that discourages misuse.

**Comment by sommerluk** [comment]
Somehow related: Rendering for #1003 is also not good.

**Comment by kocio-pl** [comment]
Could you say what dock values are acceptable for you to be rendered as water features (including name)?

**Comment by boothym** [comment]
I know what you mean about piers tagged as docks, because I fixed a few of them in the USA a while ago. But I wouldn't say actual docks are in the minority, though maybe we have differing definitions - if you could give a few examples of incorrectly tagged docks that'd be great.

Here's a few examples of docks near me - do you think these are correctly tagged as docks?
http://www.openstreetmap.org/way/467652914
http://www.openstreetmap.org/way/95068858
http://www.openstreetmap.org/way/113976843

**Comment by imagico** [comment]
Yes, in Britain the tag is fairly consistently used for water level regulated basins.  But internationally this is the exception rather than the rule.  Like with so many other tags (landuse=farm comes to mind) it was simply not foreseen when the tag was invented that it would turn out to be so ambiguous in use.

We do not need to immediately drop rendering this but i think it might be a good idea to invest in making the tagging of docks more fool proof so we have a solid basis for rendering them in a meaningful way as what they actually are.

Examples for use for non-dock water areas:

https://www.openstreetmap.org/way/186702872
https://www.openstreetmap.org/way/47864600
https://www.openstreetmap.org/way/14635282

The largest portion of incorrect uses is indeed in the US so it is possible that in the rest of the world correct use is in the majority.

**Comment by boothym** [comment]
Agree those aren't proper docks, but in the first two cases the name of the harbour or dock-like area (basin?) is something you'd want to render. Now you could just add a place=locality node to get it to show up but it would be better if you could add a way and get it rendered like named water areas.  

Your third example is apparently named "shipyard harbour" in german, not sure if that even needs to be tagged?

As for docks in the US, there's about 400 ways. Many will be proper docks, some are also tagged as piers but the rest will be mistagged. However those which are piers can be fixed with a bit of mapping time and overpass turbo :) Fortunately there's not that many new docks being wrongly added in comparison to piers.

There's also about 500 nodes, 400 of which are from an import, and they seem to be markers for harbour/basin areas:
http://www.openstreetmap.org/node/358243641
http://www.openstreetmap.org/node/358241015
http://www.openstreetmap.org/node/357732762

All of this doesn't affect the main point that proper docks are important enough to have their name rendered on the map.

**Comment by imagico** [comment]
> All of this doesn't affect the main point that proper docks are important enough to have their name rendered on the map.

This was never in doubt.  The subject of this issue is if to render all areas tagged `waterway=dock` as water areas.  We do not necessarily do those who diligently map docks a favor with that if a huge portion of uses of this tag is for other things.

Also keep in mind that rendering a floating dock as a water area is pretty pointless.

**Comment by kocio-pl** [comment]
> So i would suggest to drop rendering waterway=dockand look into if and how specific values of dock=* could be rendered, preferably in a way that discourages misuse.

Which values do you suggest?

**Comment by jeisenbe** [comment]
Any update on this issue? Should we still remove the water rendering for waterway=dock areas?

**Comment by jeisenbe** [comment]
After looking at several areas in overpass-turbo, I see that waterway=dock is that it is used for several different things, only some of which are normally areas of inland water. 

1) Many `waterway=dock` areas are part of the sea, because the tag is often used to tag a named part of a harbor, such as a slip or berth or quay, or for marines (normally tagged `leisure=marina`) This was not the original intention of use for `waterway=dock` but it is quite common (for example, see the usage around Italy). Such areas should not be rendered like `natural=water` because they should be outside of the `natural=coastline`. Also many are mapped only as a node because they do not have clearly defined outer limits, and this mapping should not be discouraged.  

2) Other `waterway=dock` features are also tagged with `man_made=pier` or are clearly meant to represent floating or elevated piers. Unfortunately this is the common meaning of the word "dock" in North American English. All the examples around Portland Oregon are this usage, for example. 

3) Dry docks and floating (dry) docks are tagged with `waterway=dock` and `dock=drydock` / `=floating`. A floating dock is certainly not an area of water, is it like a kind of barge or vessel which is used to lift boats or ships out of the water for servicing. And a dry dock appears to be dry land 99% of the time in most cases. We could considered rendering dry docks as an intermittent water area like an ephemeral lake or a flood control basin, though this might be confusing since they are rarely water filled. 

4) Tidal docks, which are also confusingly called "floating" docks or harbours in parts of the UK, fit the current wiki definition of "an enclosed area of water for ships and other craft within which the height of the water can be managed" by the use of tidal gates or locks. These can be rendered like a `water` area since they are separated from the sea. They are tagged `dock=tidal`
 
5) The term "dock" is also used for ship or boat yards, which may contain dry docks but also areas of dry land used for shipbuilding, repair and maintenance. These might better be tagged as `waterway=boatyard` or` landuse=port` or `landuse=industrial` and `industrial=shipyard` / `=port`, but some are tagged `waterway=dock` and should not be rendered as water areas. Unfortunately there is no secondary tag to look for in most cases.

**Comment by imagico** [comment]
What would you think about the idea of stopping to render waterway=dock (because it is practically used for so many different things) and start rendering `dock=drydock` and `dock=floating` in a specific styling?  That means not just like intermittent water (see #3468) but a dedicated design. `dock=drydock` seems very consistently used world wide and `dock=floating` as well - except for the US where it is sometimes used for floating piers.  A dedicated rendering that is visible even when mapped over water could help supporting mappers i think (when right now the rendering is essentially invisible).

**Comment by jeisenbe** [comment]
I agree with removing rendering for `waterway=dock` now. This would be a simple solution. 

I don’t have any particular rendering ideas for dry docks and floating dry docks at the moment, do you have any suggestions? It would need to be clearly distinct from `man_made=pier` and ideally it should not encourage mappers to use the tags inappropriately for piers. 

I wonder if the tag `dock=floating` should be supported due to the risk of confusion. It is only used 208 time on areas (and 267 on nodes), and almost all of the uses in North America are incorrect tagging for piers.

`dock=drydock` would be the best option for rendering, it is used 588 times on closed ways. Even in the USA is it mostly used for true dry docks.

**Comment by imagico** [comment]
I have not tested any specific design ideas but i think there are a lot of good options that could be tried to create a rendering that clearly hints at the nature of these and indicates that this is different from piers.

Also to consider in that context is #4514.
